docs: advertise --oci-username-variable and --oci-password-variable (#48189)
LLNL/spack: lib/spack/docs/containers.rst

   spack -e . install

   # Configure the registry
   spack -e . mirror add --oci-username-variable REGISTRY_USER \
                         --oci-password-variable REGISTRY_TOKEN \
                        container-registry oci://example.com/name/image

>    # Push the image (don't forget to set REGISTRY_USER and REGISTRY_TOKEN)

Author: ```suggestion
   # Push the image (do set REGISTRY_USER and REGISTRY_TOKEN)
```
